src/NServiceBus.Core/EndpointCreator.cs

```diff
@@ -101,13 +107,15 @@ void Initialize()
 
             recoverabilityComponent = new RecoverabilityComponent(settings);
 
-            var featureConfigurationContext = new FeatureConfigurationContext(settings, containerComponent.ContainerConfiguration,  pipelineSettings, routingComponent, receiveConfiguration);
+            var featureConfigurationContext = new FeatureConfigurationContext(settings, containerComponent.ContainerConfiguration, pipelineSettings, routingComponent, receiveConfiguration);
 
             featureComponent.Initalize(featureConfigurationContext);
             //The settings can only be locked after initializing the feature component since it uses the settings to store & share feature state.
             settings.PreventChanges();
 
-            hostingComponent = HostingComponent.Initialize(settings.Get<HostingComponent.Configuration>(), containerComponent);
+            // This needs to happen here for backwards compatibility
+            // See the test When_feature_overrides_hostinfo for more details.
+            hostingComponent.CreateHostInformation();
```

Comment:
Can I suggest to update the method name to contain "ForLegacyPurpose"? Also, if a test needs to be looked at to understand what the method is doing, perhaps the the comment should be XML doco on the `CreateHostInformation()` method?

Reply:
Good point, I've renamed to make this more clear and removed the comment